### DISCUSSION TITLE
When using GLTF Material Input -> impossible to use Colorarea

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#275] [Answers & Questions] [answered]
i can't get colorarea working when using the Material input of the GLTF scene op.

i have some 3D flags, sharing materials, hence the material input comes in handy. but now i want to tint one flag with colorarea, none of my approaches work. as if the materials are applied last in the mainloop, ignoring everything else happening in the patch.

you can test this with the help patch
![image](https://user-images.githubusercontent.com/1138673/173580900-9670a5fd-c2f5-48cb-b0f0-1cde6273bb9e.png)

when you break out objects via hierachy and place a colorarea op inbetween, nothing happens

any idea to override this behaviour ?

**Comment by pandrr** [comment, marked as answer]
1. apply colorarea under your material as you normaly would
2. get the current shader via getShader and use this as material

**Comment by wirmachenbunt** [comment]
mmm, i don't get it , here is a test scenario
https://cables.gl/p/x_syy5

**Comment by wirmachenbunt** [comment]
![image](https://user-images.githubusercontent.com/1138673/173599108-ef460d42-f328-459b-b18b-18d2c5d1720e.png)
ahso...

**Reply by wirmachenbunt** [reply]
would be nice to be able to do the same below the GLTFscene op, but i get it. its a pixelshader operation which should be applied to the material hence it doesnt work later in the flow

**Comment by pandrr** [comment]
what do you mean ? you can do it below but then you have to expose the node/hiearchy and set up the material there

**Reply by wirmachenbunt** [reply]
> what do you mean ? you can do it below but then you have to expose the node/hiearchy and set up the material there

exactly, but then i loose the material vertex selection stuff i made in blender. all is good. i understood how it works. next time i use vertex colors. makes things easier.
